Server/src/services/getDataService.ts

```diff
@@ -0,0 +1,110 @@
+        let materialRawData;
+        let yearRawData;
+        let authorRawData;
+        let categoryRawData;
+        let paramsEntered = 0;
+        // paramsEntered is used to track how many params were entered
+        let materialDatasetIds = [];
+        let yearDatasetIds = [];
+        let authorDatasetIds = [];
+        let categoryDatasetIds = [];
+
+        // Check which variables were received and the data sets IDs linked to each variable
+        if (materialReceived != undefined) {
+            paramsEntered++
+            materialRawData = await this.dataQuery.getDatasetIDFromMaterial(materialReceived);
+            materialDatasetIds = await this.createDatasetIdArray(materialRawData);
+        }
+        if (yearReceived != undefined) {
+            paramsEntered++
+            yearRawData = await this.dataQuery.getDatasetIDFromYear(yearReceived);
+            yearDatasetIds = await this.createDatasetIdArray(yearRawData);
+        }
+        if (firstNameReceived != undefined && lastNameReceived != undefined) {
+            paramsEntered++
+            authorRawData = await this.dataQuery.getDatasetIDFromAuthor(firstNameReceived, lastNameReceived);
+            authorDatasetIds = await this.createDatasetIdArray(authorRawData);
+        }
+        if (categoryReceived != undefined) {
+            paramsEntered++
+            if (subcategoryReceived != undefined) {
+                categoryRawData = await this.dataQuery.getDatasetIDFromSubcategory(categoryReceived, subcategoryReceived);
+            }
+            else {
+                categoryRawData = await this.dataQuery.getDatasetIDFromCategory(categoryReceived);
+            }
+            categoryDatasetIds = await this.createDatasetIdArray(categoryRawData);
+        }
+        let selectedDatasetIds = await this.selectDatasetIds(paramsEntered, materialDatasetIds, yearDatasetIds, authorDatasetIds, categoryDatasetIds)
+        return selectedDatasetIds
+    }
+
+    private async createDatasetIdArray(rawData: any[]) {
+        let datasetIdArray = [];
+        for (let index = 0; index < rawData.length; index++) {
+            datasetIdArray[index] = rawData[index].dataset_id;
+        }
+        return datasetIdArray;
+    }
+
+    private async selectDatasetIds(paramsEntered: number, materialDatasetIds: any[], yearDatasetIds: any[], authorDatasetIds: any[], categoryDatasetIds: any[]) {
+        let selectedDatasetIds = []
+        // Find the intersection of data sets IDs among all the different variables
+        let allDatasetIds = materialDatasetIds.concat(yearDatasetIds).concat(authorDatasetIds).concat(categoryDatasetIds)
+        let count: number
+        let currentDatasetId: number
+        for (let i = 0; i < allDatasetIds.length; i++) {
+            count = 1;
+            currentDatasetId = allDatasetIds[i]
```

Comment:
No need for this, just use the allDatasetIds[i] (as mentioned in the chat)

Reply:
Done in commit 6a0eac9